libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/data/FocalPoint.kt

```diff
@@ -0,0 +1,6 @@
+package com.mapbox.navigation.ui.maps.camera.data
+
+data class FocalPoint(
+    val x: Double,
+    val y: Double
```

Comment:
Let's check that the values are in 0 to 1 range and throw an exception if not.

Reply:
Values less than 0.0 and greater than 1.0 are also acceptable to allow puck placement outside the padded frame.

```
FocalPoint(0.5, 1.1) - will place the puck below the bottom edge of the frame (centered horizontally)
FocalPoint(0.5, -0.9) - will place the puck above the top edge of the frame (centered horizontally)
FocalPoint(-0.9, 0.5) - will place the puck to the left of the left edge of the frame (centered vertically)
FocalPoint(1.1, 0.5) - will place the puck to the right of the right edge of the frame (centered vertically)
```

<img width="418" alt="Screen Shot 2022-05-30 at 10 41 06 AM" src="https://user-images.githubusercontent.com/2678039/171015335-6d21a1bc-a23a-4bd9-a03e-89490ad407eb.png">

Reply:
When the focal point (and effectively the center of the camera) is below the padding, the whole `ViewportDataSource` logic breaks down. The class' task is to fit the provided geometry (route plus puck) into the frame (padding). When we put the focal point outside of the frame, we have no possibility to fit the geometry into the frame anymore.

In the video above, we run into the specific safeguard of the GL-Native API where geometries below the bottom edge of the padding are not framed (that's why error logs are not printed). If you would try the same if any other edge, we'd be falling back to previous valid zoom level values, but this doesn't change the fact that it makes no sense from the perspective of the `ViewportDataSource`. For example:
![Screenshot_20220530_182544_small](https://user-images.githubusercontent.com/16925074/171031728-9632f84f-eec4-49ad-b07c-a073c76f7ddf.png)
In this screenshot (`FocalPoitn(1.2, 0.8)`) we just remain in some previous fallback zoom level while not being able to frame any of the geometry (and GL-Native prints error logs on each attempt).